src/Sequence.cc

```diff
@@ -282,7 +284,37 @@ Sequence::ConstIterator Sequence::erase(Sequence::ConstIterator begin,
     return return_iter;
 }
 
-void Sequence::execute(Context& context, CommChannel* comm)
+void Sequence::execute(Context& context, CommChannel* comm_channel)
+{
+    handle_execution(context, comm_channel, "Sequence",
+        [this](Context& context, CommChannel* comm)
+        {
+            check_syntax();
+            execute_range(steps_.begin(), steps_.end(), context, comm);
+        });
+}
+
+void Sequence::execute_single_step(Context& context, CommChannel* comm_channel,
+                                   StepIndex step_index)
+{
+    if (step_index >= size())
+        throw Error(cat("Invalid step index ", step_index));
+
+    const auto step_it = steps_.begin() + step_index;
+
+    handle_execution(context, comm_channel,
+        cat("Single-step execution (", to_string(step_it->get_type()), " \"",
+            step_it->get_label(), "\")"),
+        [this, step_index, step_it](Context& context, CommChannel* comm)
+        {
+            if (executes_script(step_it->get_type()))
+                step_it->execute(context, comm, step_index);
```

Comment:
Should it not warn or something if the Step is non-executable?

Reply:
Well, it could throw. Let's discuss later if we want that. For now, I'll merge this as-is.